src/openforms/js/components/admin/form_design/variables/registration/RegistrationsSummary.js

```diff
@@ -1,19 +1,53 @@
-import React from 'react';
+import {produce} from 'immer';
+import React, {useContext, useState} from 'react';
 import {FormattedMessage, useIntl} from 'react-intl';
 
 import FAIcon from 'components/admin/FAIcon';
+import {FormContext} from 'components/admin/form_design/Context';
 import {BACKEND_OPTIONS_FORMS} from 'components/admin/form_design/registrations';
+import {SubmitAction} from 'components/admin/forms/ActionButton';
 import ButtonContainer from 'components/admin/forms/ButtonContainer';
+import SubmitRow from 'components/admin/forms/SubmitRow';
+import {FormModal} from 'components/admin/modals';
 
 /**
  *
  * @param {Object} p
  * @param {string} p.name - The name of the current registration backend
- * @param {JSX.Element} p.registrationSummary - The rendered summary of the registration backend for a specific variable
+ * @param {Object} p.variable - The current variable
+ * @param {Object} p.backend - The current backend
+ * @param {number} p.backendIndex - The current backend index (used to update the base state)
+ * @param {JSX.Element} p.registrationSummary - The rendered summary of the registration backend for the current variable
+ * @param {JSX.Element} p.variableConfigurationEditor - The rendered configuration editor for the current variable
+ * @param {() => Object} p.getOptions - The function returing the new backend options. It is provided by the configuration editor
+ * @param {(_: Object) => void} p.onChange - The base onChange function to update the base state
  * @returns {JSX.Element} - The rendered summary, containing the backend name, summary and edit icon
  */
-const RegistrationSummary = ({name, registrationSummary}) => {
+const RegistrationSummary = ({
+  name,
+  variable,
+  backend,
+  backendIndex,
+  registrationSummary,
+  variableConfigurationEditor,
+  getOptions,
+  onChange,
+}) => {
   const intl = useIntl();
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const onRegistrationOptionsChange = formData => {
+    onChange({target: `form.registrationBackend.${backendIndex}`, value: formData});
+  };
+
+  const saveRegistrationOptions = () => {
+    const newOptions = getOptions();
+    onRegistrationOptionsChange(
+      produce(backend, draft => {
+        draft.options = newOptions;
+      })
+    );
+  };
```

Comment:
I am under the impression that trying to have the save button outside the `variableConfigurationEditor` leads to really dirty code:
- It is impossible to reason about: `getOptions` is a callback set by the configuration editor, by using a `useState` hook in the parent `RegistrationsSummary` component:
https://github.com/open-formulieren/open-forms/blob/d8f15e192d03400aa0dd9d755e5140adc9be49e6/src/openforms/js/components/admin/form_design/variables/registration/RegistrationsSummary.js#L128-L130
And this is set in the configuration editor component like so:
https://github.com/open-formulieren/open-forms/blob/d8f15e192d03400aa0dd9d755e5140adc9be49e6/src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiVariableConfigurationEditor.js#L35-L41
So we end up with logic split up in three different components
- In the configuration editor, I can't make use of the `onSubmit` prop of `Formik` as the save button isn't in this component. And I am now stuck because I can't fetch the formik values outside of the `<Formik>` component

So I think it's better to have the save button explicitly handled by each backend option editor